src/App.tsx

```diff
@@ -17,18 +60,28 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                handleQuery={handleQuery}
+                handleOption={handleOption}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading ? <Loader /> : null}
+
+              <TodoList
+                todoList={list}
+                modalActive={(todo: Todo) => modalActive(todo)}
+                query={query}
+                option={option}
+                activeTaskId={task?.id || null}
+              />
```

Comment:
```suggestion
              {loading 
              ? 
              <Loader /> 
               :
              <TodoList
                todoList={list}
                modalActive={(todo: Todo) => modalActive(todo)}
                query={query}
                option={option}
                activeTaskId={task?.id || null}
              />
              }
```
please if your todos are loading - make sure to render loader, if they're finished loading, only then render a list with todos